example/protobuf-sync-message/provider/composer.json

```diff
@@ -16,11 +16,12 @@
             "rr": {
                 "version": "2023.3.10",
                 "variables": {
-                    "{$os}": "strtolower(PHP_OS_FAMILY)",
+                    "{$os}": "PHP_OS === 'Linux' && musl() === true ? 'unknown' : strtolower(PHP_OS_FAMILY)",
+                    "{$musl}": "PHP_OS === 'Linux' && musl() === true ? '-musl' : ''",
```

Comment:
(I think) I tried but I got this error:

```
FFI\Exception: Failed loading '/home/src/PhpPact/Standalone/Installer/Model/../../../../../bin/pact-ffi-lib/pact.so' (Error relocating /home/src/PhpPact/Standalone/Installer/Model/../../../../../bin/pact-ffi-lib/pact.so: __builtin_copysignq: symbol not found)
```

Reply:
it's weird that the error is coming from the pact.so and not from the rr binary.

`pact-ffi-lib/pact.so: __builtin_copysignq: symbol not found)`

Maybe we just need to try in a vanilla alpine container with arm64, download the rr binary into it and exec it. 

If not, we could exclude the rr test on musl, until this PR is able to be delivered

https://github.com/roadrunner-server/roadrunner/issues/1875#issuecomment-1966394413

we could also offer a PR to build it under qemu with docker

Reply:
So probably I did something wrong there.

Reply:
so the arm64 rr binary loads fine in an alpine container (need to find the right combo of extensions but there is actually an issue with loading the library on aarch64 musl) let me do some investigations and I'll get back to you on this.

Not sure of your hardware you have at home, but I have been testing aarch64 natively on apple silicon macbook m1 so have a bit of a quicker feedback loop.

Cheers for the efforts Tien

Reply:
So looks to be an issue with `0.4.20` version of `libpact_ffi-linux-aarch64-musl.so` 
-  (https://github.com/pact-foundation/pact-reference/releases/download/libpact_ffi-v0.4.20/libpact_ffi-linux-aarch64-musl.so.gz)

Tested with 0.4.19, results in being able to load the shared library successfully.

## 0.4.20



<img width="1801" alt="Screenshot 2024-05-14 at 18 27 29" src="https://github.com/pact-foundation/pact-php/assets/19932401/a89bb49c-55dc-45d1-a230-18cfd8958851">

## 0.4.19

<img width="2056" alt="Screenshot 2024-05-14 at 18 29 01" src="https://github.com/pact-foundation/pact-php/assets/19932401/73d925b7-8677-400b-abe7-0c3b5d051248">

I wonder if we need to pin our version of alpine to a earlier version, so that we are forward compat with as many alpine versions as possible for the shared lib.

https://github.com/pact-foundation/pact-reference/blob/e44458a8616856d3726470dd8f3884340999bf1e/rust/pact_ffi/release-linux.sh#L87-L102

Reply:
Right got a passing run of musl-arm64 local and pushed up to gh actions and passing on there. used the pre-baked build for speed (the grpc compilation takes forever in the build step even locally)

https://github.com/pact-foundation/pact-php/compare/master...YOU54F:pact-php:tx-support-alpine

content-type hack needs the condition changes to mac arm64 or windows, based on the current testing.

still need to work out why the 0.4.20 aarch64 musl so isn't ok. 

We can build the so easily from the existing a file (which is working in pact-js-core PR) so I would probably start by building the so file in an older alpine container, and seeing how it fares

Reply:
just to confirm, the none alpine variants of the road runner binary appear to work on alpine, so you can either restrict the condition to musl and x86_64, or just use the same binaries on musl/non musl

Reply:
I was a little bit confused on this one so I didn't follow you there.